doc/3.1-create-selectors-objects.md

```diff
@@ -0,0 +1,103 @@
+
+That's why selectors are objects, this way we can set/get/append properties to our selectors in order to print it out dynamically depending of Trowel variables' flags.
+
+But objects do not exists in sass : selector object are in facts maps with as key what we call a property (and as a value... its value). Those selectors objects are created and manipuled with the function `selector()`.
+
+## Create our first selector object
+To initiate a selector object, we just require to set the *block* string of our selector (remember we are strongly committed to the BEM philosophy).
+
+We will -as you can see in the snippet below- use the `selector()` function with as parameter `new` and then a string that will be our selector's block.
+
+```scss
+// initiate a selector object
+$btn--selector: selector(new 'btn');
+
+// it equals in css to :
+.btn { ... }
+```
+
+You can also create a new selector object with a specific synthax settings. It will overrule the synthax settings from the `$trowel-config` variable for this selector and its descendant.
+
+```scss
+// selector synthax options
+$btn--synthax: (
+  'prefix': 'c',
+  'separator': (
+    'prefix': '-',
+    'element': '__',
+    'modifier': '--',
+  ),
+);
+
+// initiate a selector object
+$btn--selector: selector(new 'btn' with $button--synthax);
+
+// it equals in css to :
+.c-btn { ... }
+```
+
+Notice that selectors initiated are only class. It cannot be id's or tag elements. Tag elements can be added to the class via flags.
+
+## Extend our selector object
+Our selector object is initiated and we can now extend it by adding properties that will specify more it.
+
+To set or append properties to the selector, we once again have to use the function `selector()`, and depending of its parameter, the function will return the selector object with new or edited properties.
+
+To understand how works selectors objets and the function `selector()`, try some of the functions listed below and then debug your selector variable to see how it looks like
+
+### The list of the extensions possible
+Let suppose we have a `$btn--selector` variable where we had already initiated a selector object with the function `selector(new 'btn')`.
+
+In the table below we list all the possibilty to extend this selector object. Keep in mind that the `selector()` must be stocked into a variable.
+
+| Function | role |
+| -------- | ---- |
+| `selector(set $btn--selector prefix 'd')` | set the selector prefix |
+| `selector(set $btn--selector prefix-separator '-')` | set the selector prefix separator |
+| `selector(set $btn--selector block 'btn')` | set the selector block following BEM philosophy |
+| `selector(set $btn--selector tag 'a')` | add to the selector an html tag to specify |
+| `selector(set $btn--selector pseudo-classes ('hover', 'focus'))` | add to the selector a list of pseudo-classes to specify |
```

Comment:
Same thing here, I think it would be better to have "add" or "append" instead of "set" when adding stuff to a selector